x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/crawler/crawl_custom_settings_flyout/crawl_custom_settings_flyout_mutli_crawl_scheduling.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import { EuiSpacer } from '@elastic/eui';
+
+import { CrawlCustomSettingsFlyoutCrawlDepthPanel } from './crawl_custom_settings_flyout_crawl_depth_panel';
+import { CrawlCustomSettingsFlyoutDomainsPanel } from './crawl_custom_settings_flyout_domains_panel';
+import { CrawlCustomSettingsFlyoutLogic } from './crawl_custom_settings_flyout_logic';
+import { CrawlCustomSettingsFlyoutMultiCrawlLogic } from './crawl_custom_settings_flyout_multi_crawl_logic';
+import { MultiCrawlScheduler } from './crawl_custom_settings_flyout_multi_crawl_scheduler';
+import { CrawlCustomSettingsFlyoutSeedUrlsPanel } from './crawl_custom_settings_flyout_seed_urls_panel';
+
+export const CrawlCustomSettingsFlyoutMultiCrawlScheduling: React.FC = () => {
+  const { domainUrls, multiCrawlerEntryPointUrls, multiCrawlerSitemapUrls } = useValues(
+    CrawlCustomSettingsFlyoutLogic
+  );
+
+  const {
+    crawlerConfigurations,
+    crawlerConfigActiveTab,
+    index: crawlerIndex,
+  } = useValues(CrawlCustomSettingsFlyoutMultiCrawlLogic);
+
+  const {
+    onSelectMaxCrawlDepth,
+    onSelectDomainUrls,
+    onSelectCustomEntryPointUrls,
+    onSelectCustomSitemapUrls,
+    onSelectEntryPointUrls,
+    onSelectSitemapUrls,
+    toggleIncludeSitemapsInRobotsTxt,
+    setConnectorSchedulingInterval,
+    onSetConnectorSchedulingEnabled,
+  } = useActions(CrawlCustomSettingsFlyoutMultiCrawlLogic);
+
+  console.log(crawlerConfigurations);
+
+  return crawlerConfigurations.map((config, index) => {
+    if (index === crawlerConfigActiveTab) {
+      return (
+        <React.Fragment key={index}>
+          <CrawlCustomSettingsFlyoutCrawlDepthPanel
+            maxCrawlDepth={config.maxCrawlDepth}
+            onSelectMaxCrawlDepth={(e) => onSelectMaxCrawlDepth(index, e)}
+          />
+          <EuiSpacer />
+          <CrawlCustomSettingsFlyoutDomainsPanel
+            selectedDomainUrls={config.selectedDomainUrls}
+            domainUrls={domainUrls}
+            onSelectDomainUrls={(e) => onSelectDomainUrls(index, e)}
+          />
+          <EuiSpacer />
+          <CrawlCustomSettingsFlyoutSeedUrlsPanel
+            customEntryPointUrls={config.customEntryPointUrls}
```

Comment:
Nitpick(non-blocking): Maybe you can pass config directly as it would be the same(or similar enough) for single or multiple crawls. Plus you already have a type for it

Reply:
Thank you for the suggestion, it made the code much cleaner!